cognite/client/data_classes/agents/chat.py

```diff
@@ -85,6 +86,194 @@ def _load_content(cls, data: dict[str, Any]) -> UnknownContent:
+        return {
+            "type": self._type,
+            "clientTool": {
+                "name": self.name,
+                "description": self.description,
+                "parameters": self.parameters,
+            },
+        }
+
+    @classmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ClientToolAction:
+        client_tool = data["clientTool"]
+        return cls(
+            name=client_tool["name"],
+            description=client_tool["description"],
+            parameters=client_tool["parameters"],
+        )
+
+
+@dataclass
+class UnknownAction(Action):
+    """Unknown action type for forward compatibility.
+
+    Args:
+        type (str): The action type.
+        data (dict[str, Any]): The raw action data.
+    """
+
+    type: str
+    data: dict[str, Any] = field(default_factory=dict)
+
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        result = self.data.copy()
+        result["type"] = self.type
+        return result
+
+    @classmethod
+    def _load_action(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> UnknownAction:
+        action_type = data.get("type", "")
+        return cls(data=data, type=action_type)
+
+
+# Build the mapping AFTER concrete classes are defined
+_ACTION_CLS_BY_TYPE: dict[str, type[Action]] = {
+    subclass._type: subclass  # type: ignore[type-abstract]
+    for subclass in Action.__subclasses__()
+    if hasattr(subclass, "_type") and not getattr(subclass, "__abstractmethods__", None)
+}
+
+
+@dataclass
+class ActionCall(CogniteObject, ABC):
+    """Base class for action calls requested by the agent."""
+
+    _type: ClassVar[str]
+    action_id: str
+
+    @classmethod
+    def _load(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ActionCall:
+        """Dispatch to the correct concrete action call class based on `type`."""
+        action_type = data.get("type", "")
+        action_class = _ACTION_CALL_CLS_BY_TYPE.get(action_type, UnknownActionCall)
+        return action_class._load_call(data, cognite_client)
+
+    @abstractmethod
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        """Dump the action call to a dictionary."""
+        ...
+
+    @classmethod
+    @abstractmethod
+    def _load_call(cls, data: dict[str, Any], cognite_client: CogniteClient | None = None) -> ActionCall:
+        """Create a concrete action call instance from raw data."""
+        ...
+
+
+@dataclass
+class ClientToolCall(ActionCall):
+    """A client tool call requested by the agent.
+
+    Args:
+        action_id (str): The unique identifier for this action call.
+        name (str): The name of the client tool being called.
+        arguments (dict[str, Any]): The parsed arguments for the tool call.
+    """
+
+    _type: ClassVar[str] = "clientTool"
+    action_id: str
+    name: str
+    arguments: dict[str, Any]
```

Comment:
Just a check. You know that `arguments` is always `json` serializable? In the API spec you sent this is string.